### PR TITLE
Use query builder instead of OC_DB in trashbin

### PR DESCRIPTION
**Pull request by nickvergessen**
Ref #23929

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Review by icewind1991 — APPROVED** [review]
:tada:

File: apps/files_trashbin/lib/Trashbin.php [hunk 1]
```diff
@@ -208,8 +219,13 @@ private static function copyFilesToUser($sourcePath, $owner, $targetPath, $user,
 
 
 		if ($view->file_exists($target)) {
-			$query = \OC_DB::prepare("INSERT INTO `*PREFIX*files_trash` (`id`,`timestamp`,`location`,`user`) VALUES (?,?,?,?)");
-			$result = $query->execute([$targetFilename, $timestamp, $targetLocation, $user]);
+			$query = \OC::$server->getDatabaseConnection()->getQueryBuilder();
+			$query->insert('files_trash')
+				->set('id', $query->createNamedParameter($targetFilename))
```
Comment (MorrisJobke) [inline apps/files_trashbin/lib/Trashbin.php:224]:
```suggestion
				->setValue('id', $query->createNamedParameter($targetFilename))
```

It's `setValue` or `values`.

**Review by MorrisJobke — APPROVED** [review]
Fixed, tested and works 👍

**Comment by skjnldsv** [comment]
comments failure unrelated

**Pull request merged**